src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -3,19 +3,162 @@
+    public void shouldDeleteProduct() {
+        Long id = 1L;
+
+        productDao.delete(id);
+
+        long count = productDao
+                .getProducts()
+                .stream()
+                .filter(p -> id.equals(p.getId()))
+                .count();
+        assertEquals(0L, count);
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public void shouldSaveNull() {
+        productDao.save(null);
+    }
+
+    @Test
+    public void shouldProductWithNullId() {
+        notDefaultProduct.setId(null);
+
+        productDao.save(notDefaultProduct);
+
+        long count = productDao
+                .getProducts()
+                .stream()
+                .filter(p -> notDefaultProduct.getDescription().equals(p.getDescription()))
+                .filter(p -> p.getId() != null)
+                .count();
+        assertEquals(1L, count);
+    }
+
+    @Test
+    public void shouldSaveNewProduct() {
+        productDao.save(notDefaultProduct);
+
+        long count = productDao
+                .getProducts()
+                .stream()
+                .filter(p -> notDefaultProduct.getDescription().equals(p.getDescription()))
+                .count();
+        assertEquals(1l, count);
+    }
+
+    @Test
+    public void shouldSaveUpdateProduct() {
+        Long existId = 1L;
+        notDefaultProduct.setId(existId);
+
+        productDao.save(notDefaultProduct);
+
+        Product product = productDao
+                .getProducts()
+                .stream()
+                .findAny()
+                .get();
+        assertEquals("iphone7", product.getCode());
+        assertEquals(existId, product.getId());
+    }
+
+    @Test
+    public void shouldSaveProductWithId() {
+        Long id = 17L;
+        notDefaultProduct.setId(id);
+
+        productDao.save(notDefaultProduct);
+
+        Product product = productDao
+                .getProducts()
+                .stream()
+                .filter(p -> id.equals(p.getId()))
+                .findAny()
+                .get();
+        assertEquals(notDefaultProduct, product);
+    }
+
+    @Test
+    public void shouldGetSize() {
+        assertEquals(13, productDao.getSize());
+    }
+
+    @Test
+    public void shouldGetSizeSaveProduct() {
+        productDao.save(notDefaultProduct);
+
+        assertEquals(14, productDao.getSize());
+    }
+
+    @Test
+    public void shouldGetSizeDeleteProduct() {
+        productDao.delete(1L);
+
+        assertEquals(12, productDao.getSize());
+    }
+
+
+    @Test
+    public void shouldDeleteAll() {
+        productDao.deleteAll();
+
+        assertEquals(0, productDao.getSize());
+        assertTrue(((Long) 0L).equals(productDao.getNextId()));
+    }
+
+    @Test
+    public void shouldFindProductsNoResultsByStock() {
+        productDao.deleteAll();
+        notDefaultProduct.setStock(0);
+        productDao.save(notDefaultProduct);
+
+        assertTrue(productDao.findProducts().isEmpty());
```

Comment:
The same. Ex:
//when part

var result = productDao.findProducts();

assertTrue(result.isEmpty());


Please fix this issue for the entire class.